src/decorators.ts

```diff
@@ -126,6 +126,30 @@ export function ResponseSchema(
         },
       }
 
+      const oldSchema =
+        source.responses[statusCode]?.content[contentType].schema
+      if (oldSchema?.$ref || oldSchema?.items?.$ref) {
+        // case where we're adding multiple schemas under single statuscode/contentType
+        // with single $ref
+        const isOldSchemaArray = oldSchema?.items?.$ref
+
+        // delete old schema and integrate into current schema under oneOf
+        const schemaObj = { oneOf: [{ ...oldSchema }, schema] }
+        responses[statusCode].content[contentType].schema = schemaObj
+        
+        if (isOldSchemaArray) {
+          delete oldSchema.items
+          delete oldSchema.type
+        } else {
+          delete oldSchema.$ref
+        }
+      } else if (oldSchema?.oneOf) {
+        // case where there's already multiple existing schemas
+        const oneOf = _.concat([...oldSchema.oneOf], schema)
+        responses[statusCode].content[contentType].schema = { oneOf }
+        delete oldSchema.oneOf
+      }
+
       return _.merge({}, source, { responses })
```

Comment:
I'm thinking all the `delete`s makes this a bit hard to read. How about splitting into two branches, something like
```typescript
if (oldSchema) {
  const newSchema = {};

  // ...construct newSchema

  source.responses[statusCode].content[contentType].schema = newSchema
  return source
} else {
  return _.merge({}, source, { responses }) // just as before
}
```

Reply:
addressed [here](https://github.com/epiphone/routing-controllers-openapi/pull/58/commits/c67e5e8acfdeb4dda460eef14bc4b3b4aa41f435)